spring-rabbit/src/test/java/org/springframework/amqp/rabbit/AsyncRabbitTemplateTests.java

```diff
@@ -394,6 +396,29 @@ public void testStopCancelled() throws Exception {
 		assertThat(callback.result).isNull();
 	}
 
+	@Test
+	@DirtiesContext
```

Comment:
I wonder if we really need this `@DirtiesContext`.
I see that many other tests in this class have it, but that doesn't mean that it is right.
Would you mind to double check if it works without?
Just possible slight optimization in execution performance!

Reply:
Every test changes the `AsyncRabbitTemplate` state, so we must reset the context before each test. 
In my test, for example, I'm changing the messageConverter on the `RabbitTemplate`, so if my test will run first, all other tests will fail.